server/controllers/submission.js

```diff
@@ -16,13 +18,35 @@ exports.getAllSubmissions = asyncHandler(async (req, res, next) => {
   }
 
   try {
-    const isOwner = await Contest.fineOne({ userId: userId });
+    const isOwner = await Contest.findOne({ userId: userId });
     if (isOwner) {
-      const submissions = Submission.find({ contestId: contestId });
-      res.status(200).json({ submission: submissions });
+      const submissions = await Submission.find({ contestId: contestId });
```

Comment:
As each submission has a userId, maybe populate the user document here and select the fields you need?

Reply:
optimize the controller